tests/test_markdown.py

```diff
@@ -614,6 +614,16 @@ def test_link_with_title(markdown_function, expected):
     assert markdown_function('[Example](http://example.com "An example URL")') == expected
 
 
+def test_letter_qr_code():
+    expected_qr_code_start = "<p><div class='qrcode'><svg viewBox=\"0 0 25 25\">"
+    assert notify_letter_preview_markdown('[qr](http://example.com")').startswith(expected_qr_code_start)
+
+
+def test_letter_qr_code_works_with_extra_whitespace():
+    expected_qr_code_start = "<p><div class='qrcode'><svg viewBox=\"0 0 25 25\">"
+    assert notify_letter_preview_markdown('[ qr ](http://example.com")').startswith(expected_qr_code_start)
```

Comment:
I think I'd quite like to test the entire QR code SVG remains the same as I think it should be a static value. Or does it change every time you generate it?

Gives us a bit more confidence+warning if (for example) we update the library and the implementation changes under the hood.

In the mid/long-term I'd like a functional test that generates a letter with a QR code, and then scans the QR code to validate the URL is expected. This would give us end-to-end confirmation that the QR is generated correctly and readable.

The reason I'm thinking we need some changes like this are because this test (as it stands) wouldn't detect that  segno QR code was broken by the markdown renderer replacing hyphens with designer hyphens.

Reply:
Okee I updated the first test 😁